loris/resolver.py

```diff
@@ -340,14 +368,36 @@ def copy_to_cache(self, ident):
                 rename(tmp_file.name, local_fp)
                 logger.info("Copied %s to %s", source_url, local_fp)
 
+        # Check for rules file associated with image file
+        # These files are < 2k in size, so fetch in one go.
+        # Assumes that the rules will be next to the image
+        # cache_dir is image specific, so this is easy
+
+        bits = split(source_url)
+        fn = bits[1].rsplit('.')[0] + "." + self.auth_rules_ext
```

Comment:
Seems like source_url would be something like http://mysite.com/images/foo.jpg, so split() would just return ['http://mysite.com/images/foo.jpg'], and bits[1] would give an IndexError. We could factor out a _rules_url(source_url) method and test it with various inputs, and make sure it's returning what we want.